src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,197 @@
+     * @throws DukeException Throws a DukeException.
+     */
+    public void parse(String cmd) throws DukeException {
+        String[] temp = cmd.split(" ");
+        String mainCmd = temp[0];
+        String[] subCmd = Arrays.copyOfRange(temp, 1, temp.length);
+
+        switch (mainCmd) {
+        case "list":
+            this.parseList(subCmd);
+            break;
+        case "mark":
+            this.parseMark(subCmd);
+            break;
+        case "unmark":
+            this.parseUnmark(subCmd);
+            break;
+        case "todo":
+            this.parseTodo(subCmd);
+            break;
+        case "deadline":
+            this.parseDeadline(subCmd);
+            break;
+        case "event":
+            this.parseEvent(subCmd);
+            break;
+        case "delete":
+            this.parseDelete(subCmd);
+            break;
+        case "find":
+            this.parseFind(subCmd);
+            break;
+        default:
+            throw new InvalidDescriptionException();
+        }
+    }
+
+    /**
+     * Parses the list command.
+     * @param subCmd An Array of Strings containing the remaining command arguments.
+     * @throws InvalidDescriptionException Throws an invalidDescriptionException.
+     */
+    public void parseList(String[] subCmd) throws InvalidDescriptionException {
+        if (subCmd.length != 0) {
+            throw new InvalidDescriptionException();
+        } else {
+            this.taskList.listTasks();
+        }
+    }
+
+    /**
+     * Parses the mark command.
+     * @param subCmd An Array of Strings containing the remaining command arguments.
+     * @throws DukeException Throws a DukeException.
+     */
+    public void parseMark(String[] subCmd) throws DukeException {
+        if (Integer.parseInt(subCmd[0]) <= 0 || Integer.parseInt(subCmd[0]) > this.taskList.size()) {
+            throw new InvalidDescriptionException();
+        } else {
+            this.taskList.getTask(Integer.parseInt(subCmd[0]) - 1).mark();
+            this.storage.save(this.taskList);
+        }
+    }
+
+    /**
+     * Parses the unmark command.
+     * @param subCmd An Array of Strings containing the remaining command arguments.
+     * @throws DukeException Throws a DukeException.
+     */
+    public void parseUnmark(String[] subCmd) throws DukeException {
+        if (Integer.parseInt(subCmd[0]) <= 0 || Integer.parseInt(subCmd[0]) > this.taskList.size()) {
+            throw new InvalidDescriptionException();
+        } else {
+            this.taskList.getTask(Integer.parseInt(subCmd[0]) - 1).unmark();
+            this.storage.save(this.taskList);
+        }
+    }
+
+    /**
+     * Parses the todo command.
+     * @param subCmd An Array of Strings containing the remaining command arguments.
+     * @throws DukeException Throws a DukeException.
+     */
+    public void parseTodo(String[] subCmd) throws DukeException {
+
+        String tmp = String.join(" ", subCmd);
```

Comment:
Do you think that a better naming for this variable can be _fullDescription_ instead of _tmp_ ?